engine/table/src/main/java/io/deephaven/engine/table/impl/naturaljoin/IncrementalNaturalJoinStateManagerTypedBase.java

```diff
@@ -286,6 +301,8 @@ public boolean doRehash(boolean fullRehash, MutableInt rehashCredits, int nextCh
     }
 
     protected void newAlternate() {
+        alternateEntries = numEntries;
```

Comment:
I would argue for in-lining all of this in `doRehash`, making `newAlternate()` just `abstract`. `clearAlternate()` has the same (less than ideal) pattern.